math/mathcore/inc/Math/IFunction.h

```diff
@@ -346,6 +362,16 @@ namespace ROOT {
                grad[icoord] = BaseGrad::Derivative(x, icoord);
          }
 
+         /// In some cases, the gradient algorithm will use information from the previous step, these can be passed
+         /// in with this overload. The `previous_*` arrays can also be used to return second derivative and step size
+         /// so that these can be passed forward again as well at the call site, if necessary.
+         virtual void Gradient(const T *x, T *grad, T *previous_grad, T *previous_g2, T *previous_gstep) const
```

Comment:
```suggestion
         virtual void GradientWithPrevResult(const T *x, T *grad, T *previous_grad, T *previous_g2, T *previous_gstep) const
```
See equivalent comment about `DoGradient`.